src/qmapshack/poi/CRawPoi.cpp

```diff
@@ -19,63 +19,344 @@
+    QString addr = QObject::tr("addr:housenumber addr:street<br>\naddr:city addr:postcode<br>\naddr:country");
+    static const QRegularExpression addressRegex("(addr:[A-z]+)", QRegularExpression::MultilineOption);
+    QRegularExpressionMatchIterator i = addressRegex.globalMatch(addr);
+    while (i.hasNext())
     {
-        return category;
+        const QRegularExpressionMatch& match = i.next();
+        const QString& key = match.captured(1);
+        if(data.contains(key))
+        {
+            addr.replace(key, data[key]);
+            tempSkipKeys.append(key);
+        }
+        else
+        {
+            addr.replace(key, "");
+        }
     }
-    return name;
+    desc += addr;
+
+    if(nameRelatedKeys.count() > 1)
+    {
+        if(!desc.isEmpty())
+        {
+            desc += "<br>\n";
+        }
+        desc += "<b>" + QObject::tr("Alternative names: ") + "</b>";
+        bool firstName = true;
+        for(const QString& key : nameRelatedKeys)
+        {
+            if(data[key] == name)
+            {
+                continue;
+            }
+
+            if(firstName)
+            {
+                firstName = false;
+            }
+            else
+            {
+                desc += ", ";
+            }
+            desc += data[key];
+        }
+    }
+    if(data.contains("phone"))
+    {
+        if(!desc.isEmpty())
+        {
+            desc += "<br>\n";
+        }
+        desc += "<b>" + QObject::tr("Phone number: ") + "</b>" + data["phone"];
+    }
+    if(data.contains("contact:phone"))
+    {
+        if(!desc.isEmpty())
+        {
+            desc += "<br>\n";
+        }
+        desc += "<b>" + QObject::tr("Phone number: ") + "</b>" + data["contact:phone"];
+    }
+    if(data.contains("contact:mobile"))
+    {
+        if(!desc.isEmpty())
+        {
+            desc += "<br>\n";
+        }
+        desc += "<b>" + QObject::tr("Mobile phone number: ") + "</b>" + data["contact:mobile"];
+    }
+    if(!desc.isEmpty())
+    {
+        desc += "<br>\n";
+    }
+    // Append all other tags
+    bool firstTag = true;
+    for(const QString& key : data.keys())
+    {
+        if(skipKeys.contains(key)
+           || tempSkipKeys.contains(key)
+           || wikidataRelatedKeys.contains(key)
+           || wikipediaRelatedKeys.contains(key)
+           || nameRelatedKeys.contains(key)
+           || key == "phone"
+           || key == "contact:phone"
+           || key == "contact:mobile")
+        {
+            continue;
+        }
+
+        if(firstTag)
+        {
+            firstTag = false;
+        }
+        else
+        {
+            desc += "<br>\n";
+        }
+        desc += key + ": " + data[key];
+    }
+
+    return desc;
+}
+
+QList<IGisItem::link_t> CRawPoi::getLinks() const
+{
+    QList<IGisItem::link_t> links;
+    if(data.contains("contact:email"))
+    {
+        links += {"mailto:" + data["contact:email"], QObject::tr("Email"), "text/html"};
```

Comment:
Please no `QObject::tr()` Use `Q_DECLARE_TR_FUNCTIONS` macro to define a local `tr()`

Reply:
I knew there was some other way to do it :thinking:

Reply:
Yes and by that Linguist is listing the stuff as part of CRawPoi helping the translators to find the correct context.